lib/components/segmented-control/base.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { KeyboardEvent, RefObject, useEffect, useState } from "react";
+import { TRegularSizes } from "../../types";
+import { Segment } from "./segment";
+import "./segmented-control.scss";
+
+interface Option {
+    disabled?: boolean;
+    icon?: string | React.ReactNode;
```

Comment:
@maryia-deriv we can pass icon as a quill Icon component` icon ? : QuillIconComponent`

export type QuillIconComponent = React.ForwardRefExoticComponent<
    Omit<QuillSvgProps, "ref">>;
 
and use it as a component like `<Icon iconSize={} className={}/>`

Then we can pass it in story book as `icon : LabelPairedPlaceholderMdRegularIcon`

Reply:
@balakrishna-deriv I had to add support for 'placeholder' because of a known issue with storybook documentation https://github.com/storybookjs/storybook/issues/17098, which gets stuck in an infinite loop when you try passing an icon as a component from within a story.